sphinx/ext/autosectionlabel.py

```diff
@@ -29,6 +29,17 @@ def get_node_depth(node: Node) -> int:
     return i
 
 
+def get_node_parents(node: Node) -> list[str]:
+    parents = []
+    cur_node = node
+    while cur_node.parent != node.document:
+        cur_node = cur_node.parent
+        title = cast(nodes.title, cur_node[0])
+        ref_name = getattr(title, 'rawsource', title.astext())
+        parents.append(ref_name)
+    return parents
```

Comment:
This will not be sufficient. You should walk the parent nodes that are only `nodes.section` and not other kind of nodes that could appear.

Reply:
I am very new to Sphinx, can you give me an example, please?

Reply:
Sure! I'll try to see if what I had in mind works tomorrow. If not, your algorithm would be correct (and we could also add `assert isinstance(node.parent, nodes.section)`)